src/pyproject.toml.m4

```diff
@@ -1,29 +1,161 @@
+include(`sage_spkg_versions_toml.m4')dnl' -*- conf-toml -*-
 [build-system]
 # Minimum requirements for the build system to execute.
 requires = [
     "sage_setup[autogen]",
     # Some version of sage-conf is required.
     # Note that PEP517/518 have no notion of optional sage_spkg dependencies:
     # https://github.com/pypa/pip/issues/6144
-     esyscmd(`sage-get-system-packages install-requires-toml \
-        sage_conf      \
-        setuptools \
-        wheel          \
-        sage_setup     \
-        cypari         \
-        cysignals      \
-        cython         \
-        gmpy2          \
-        jinja2         \
-        jupyter_core   \
-        numpy          \
-        pkgconfig      \
-        pplpy          \
-        memory_allocator \
-                    ')]
+    SPKG_INSTALL_REQUIRES_sage_conf
+    SPKG_INSTALL_REQUIRES_setuptools
+    SPKG_INSTALL_REQUIRES_wheel
+    SPKG_INSTALL_REQUIRES_sage_setup
+    SPKG_INSTALL_REQUIRES_cypari
+    SPKG_INSTALL_REQUIRES_cysignals
+    SPKG_INSTALL_REQUIRES_cython
+    SPKG_INSTALL_REQUIRES_gmpy2
+    SPKG_INSTALL_REQUIRES_jupyter_core
+    SPKG_INSTALL_REQUIRES_memory_allocator
+    SPKG_INSTALL_REQUIRES_numpy
+    SPKG_INSTALL_REQUIRES_pkgconfig
+    SPKG_INSTALL_REQUIRES_pplpy
```

Comment:
There is a problem here.  All of those are not build-system requirements. According to pep 517 this should only list packages that are needed to run the build-system (not to build a wheel).

Only sage-setup and setuptools are necessary to run the build system, and indeed they suffice to build a sdist.

Reply:
I salute you for starting to read the specs.

But you're a little bit too quick to declare that this "needs work".

There is not "one true" declaration of build-system requirements. 

As I explained to you elsewhere, setuptools does *not* support declarations of PEP-517/660 hooks (other backends do). When we decide to use setuptools as the PEP-517/660 build-backend, we work with what is supported. We have to build wheels, the wheels need all of these things, hence we declare them as build-system requirements.

Reply:
Besides, the present PR does not even make a change to what is declared...